### PR TITLE
Refactor e2e tests to adhere to server-side container ID

### PR DESCRIPTION
**Pull request by alpaix**
Enforce the server-side container ID generation in the r11s driver and refactor end-to-end tests to adopt the new create container flow.
The main idea behind the refactoring is to make the test provider record the actual document ID of the newly created container that can be used in subsequent load container calls.

### CODE REVIEW

File: packages/drivers/routerlicious-driver/src/documentServiceFactory.ts
Comment (curtisman) [inline packages/drivers/routerlicious-driver/src/documentServiceFactory.ts:81]: Does this work for t9s and r11s?

File: packages/drivers/routerlicious-driver/src/documentServiceFactory.ts
Comment (znewton) [inline packages/drivers/routerlicious-driver/src/documentServiceFactory.ts:67]: Until the example apps like Clicker work with server-side document id, I'm not sure we should enforce it in the driver. What is the benefit of doing this client-side enforcing now?

File: packages/drivers/routerlicious-driver/src/documentServiceFactory.ts
Comment (znewton) [reply packages/drivers/routerlicious-driver/src/documentServiceFactory.ts:81]: Yes, both have support for it after #7022 + #7038 (for R11s) and #7104 (for T9s)

File: packages/drivers/routerlicious-driver/src/documentServiceFactory.ts
Comment (alpaix) [reply packages/drivers/routerlicious-driver/src/documentServiceFactory.ts:67]: Needed to make sure tests are valid and consistent. Examples will be fixed shortly after.

File: packages/test/test-utils/src/testObjectProvider.ts
Comment (alpaix) [reply packages/test/test-utils/src/testObjectProvider.ts:206]: Good idea! This refactoring might take longer time. My plan was to unblock tests and move on to fixing examples. We can get back to revisiting test object provider once we're done with the container ID work. 

Does it make sense?

File: packages/drivers/routerlicious-driver/src/restWrapper.ts
Comment (alpaix) [inline packages/drivers/routerlicious-driver/src/restWrapper.ts:32]: The client build started to fail after some protocol changes got pushed to the main. Specifically #7522

File: packages/drivers/routerlicious-driver/src/documentServiceFactory.ts
Comment (znewton) [reply packages/drivers/routerlicious-driver/src/documentServiceFactory.ts:67]: If this merges before the apps, then Clicker will become unusable until the app changes are in. Is it possible to merge the app changes first? We use Clicker very frequently for testing Azure Fluid Relay, so breaking the latest build with this change is not a good spot to be in.

Is my understanding correct that the tests will always pass in a documentId, so this change is required to make the driver ignore the test's generated document id and actually validate server-side ids?

Comment (alpaix) [reply packages/drivers/routerlicious-driver/src/documentServiceFactory.ts:67]: Agreed. Apps should go first still in the back-compat mode. Meaning, not passing the document ID but the driver remains compatible with tests.

Comment (alpaix) [reply packages/drivers/routerlicious-driver/src/documentServiceFactory.ts:67]: Regarding tests, they will keep sending the document ID for the back-compat validation, e.g. against 0.39 driver.

File: packages/test/test-utils/src/testObjectProvider.ts
Comment (curtisman) [reply packages/test/test-utils/src/testObjectProvider.ts:206]: yep, but please make sure that this will work for all real service target (including t9s, r11s, frs, and odsp)
Please open an issue to track our plan to refactor this better.

File: packages/drivers/routerlicious-driver/src/documentServiceFactory.ts
Comment (alpaix) [reply packages/drivers/routerlicious-driver/src/documentServiceFactory.ts:67]: Example apps are supported via the latest webpack loader merged earlier.

**Comment by alpaix** [comment]
@curtisman @znewton All tests passed, webpack loader refactored, the pr is ready to go. Can I get an approval?

**Review by curtisman — APPROVED** [review]
:shipit:

**Pull request merged**
